### PR TITLE
[CALCITE-6655] Aggregation of deeply nested window not detected when unparsing

### PR DESCRIPTION
**Pull request by linorosa**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 1]
```diff
@@ -2122,6 +2123,28 @@ private boolean hasNested(
       return false;
     }
 
+    /** Recursively looks for a node matching the operandPredicate. */
```
Comment (mihaibudiu) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java:2126]: this is usually done using a visitor patter, see SqlShuttle.

Comment (linorosa) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: ahh thanks for the feedback @mihaibudiu. I just pushed a refactor to use an SqlShuttle at #
226b7d3

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -8741,6 +8741,48 @@ private void checkLiteral2(String expression, String expected) {
         .dialect(MssqlSqlDialect.DEFAULT).ok(mssqlExpected);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6655">[CALCITE-6655]
+   * Aggregation of deeply nested aggregations </a>.
```
Comment (mihaibudiu) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:8746]: extra space before closing tag
You need to use the exact same message as in the JIRA case.
Please use this for your PR as well.
These conventions help us keep track of the correspondence between commits, PRs, and JIRA cases.

**Review by mihaibudiu — APPROVED** [review]
The squashing is done usually after the changes have accepted.
Code review may be easier if the reviewer can see separate commits addressing the suggestions.
So, yes, please go ahead and squash the commits in preparation for merging.

**Pull request merged**